core-backend/src/runtime.rs

```diff
@@ -71,6 +77,23 @@ where
     pub fn ext_mut(&mut self) -> &mut Ext {
         &mut self.state_mut().ext
     }
+
+    pub fn set_io(
+        &mut self,
+        io: Result<MemoryAccessIo<Caller, BackendMemory<ExecutorMemory>>, MemoryAccessError>,
+    ) {
+        self.io = Some(io);
+    }
+
+    pub fn io_ref(
+        &self,
+    ) -> Result<&MemoryAccessIo<Caller, BackendMemory<ExecutorMemory>>, MemoryAccessError> {
+        match &self.io {
+            Some(Ok(io)) => Ok(io),
+            Some(Err(err)) => Err(err.clone()),
+            None => Err(MemoryAccessError::ManagerError),
```

Comment:
```suggestion
            None => unreachable!(),
```
Syscall didn't set its requirement for the manager and thus implementation is invalid and should be never met in runtime

Reply:
Agree